src/main/fc/boot.c

```diff
@@ -723,9 +733,32 @@ void configureScheduler(void)
 {
     schedulerInit();
     setTaskEnabled(TASK_SYSTEM, true);
+
+    rescheduleTask(TASK_GYROPID, gyro.targetLooptime);
     setTaskEnabled(TASK_GYROPID, true);
-    rescheduleTask(TASK_GYROPID, imuConfig()->gyroSync ? targetLooptime - INTERRUPT_WAIT_TIME : targetLooptime);
-    setTaskEnabled(TASK_ACCEL, sensors(SENSOR_ACC));
+
+    if (sensors(SENSOR_ACC)) {
+        switch (gyro.targetLooptime) {  // Switch statement kept in place to change acc rates in the future
```

Comment:
@hydra: I was commenting `switch` statement only. If `gyro.targetLooptime` is not exactly 8000/<small integer>, it won't work.
Using series of if/else will fix this